src/object/transform.c

```diff
@@ -413,6 +413,15 @@ static CT_ATTR ct_partition_atts[] = {
   {"comment", NULL_ATTRID, DB_TYPE_VARCHAR}
 };
 
+static CT_ATTR ct_synonym_atts[] = {
+  {"synonym_name", NULL_ATTRID, DB_TYPE_VARCHAR},
+  {"synonym_owner_name", NULL_ATTRID, DB_TYPE_VARCHAR},
```

Comment:
Wouldn't it be better to specify db_user instead of a string?
`{"synonym_owner_name", NULL_ATTRID, DB_TYPE_OBJECT},`

When creating it, you can do it like the code below.
`error_code = smt_add_attribute(def, "synonym_owner_name", AU_USER_CLASS_NAME, NULL);`

Reply:
I wanted to avoid using the db_user type because it is OID dependent.

If the synonym_owner_name column type becomes db_user type,
the user must write a subquery for db_user and use the where condition.

Even if it is not an OID value, the owner_name is also a unique value,
so it is not necessary to set it as a db_user type.

Reply:
It seems that DB_TYPE_OBJECT type is used to have the same effect on columns that require foreign key setting in system tables.

This is because when a foreign key is set,
the order of system table creation and deletion depends on the foreign key.

If so, it seems appropriate to use the DB_TYPE_OBJECT type.

Reply:
It seems that DB_TYPE_OBJECT type is used to have the same effect on columns that require foreign key setting in system tables.

This is because when a foreign key is set,
The order of system table creation and deletion is affected by foreign keys.